packages/ui/src/apis/interaction.ts

```diff
@@ -200,3 +200,21 @@ export const bindSocialRelatedUser = async (payload: SocialEmailPayload | Social
 
   return api.post(`${interactionPrefix}/submit`).json<Response>();
 };
+
+export const linkWithSocial = async (connectorId: string) => {
+  // Replace email/phone profile with connectorId. User previous verified email/phone sign-in instead
+
+  await api.put(`${interactionPrefix}/event`, {
+    json: {
+      event: InteractionEvent.SignIn,
+    },
+  });
+
+  await api.put(`${interactionPrefix}/profile`, {
+    json: {
+      connectorId,
+    },
+  });
+
+  return api.post(`${interactionPrefix}/submit`).json<Response>();
```

Comment:
Calling 3 async APIs in a row might be a bit low efficient, so can we combine the first 2 together into a new one, or at least use `Promise.all()`?

Reply:
nope, have to run sync. That is the side effect caused by splitting interaction APIs into restful ones.

Reply:
OK, then can we put both event and profile in one request?

Reply:
discussed offline